integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/work/IndexingFieldTypesIT.java

```diff
@@ -102,6 +102,19 @@ public void withReference() {
 
 			assertThatQuery( query ).hasHitsAnyOrder( expectedDocuments );
 		}
+
+		for ( int i = 0; i < values.size(); i++ ) {
+			SearchQuery<IdAndValue<F>> query = scope.query()
+					.select( f -> f.composite(
+							(id, val) -> new IdAndValue<>( id, val ),
+							f.id( String.class ),
+							f.field( absoluteFieldPath, typeDescriptor.getJavaType() )
+					) )
+					.where( f -> f.matchAll() )
+					.toQuery();
+
+			assertThatQuery( query ).hasHitsAnyOrder( expectedDocuments );
+		}
```

Comment:
I think you can remove the block above with the entity reference projection. This test is not about testing the entity reference projection, we're just using it because it's convenient.

But it's a good idea to use the ID projection instead, now that we have it.